File: nj2k/testData/newJ2k/nullability/Overrides.java
//file
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

class Base {
    @Nullable
    public String foo(@Nullable String s) { return ""; }

    public String bar(String s) {
        return s != null ? s + 1 : null;
    }

    public String zoo(Object o){ return ""; }

    public String nya(String s) { return s; }
}

interface I {
    @Nullable String zoo(@Nullable Object o);

    public String nya(String s) { return ""; }
}

class C extends Base implements I {
    public String foo(String s) { return ""; }

    public String bar(String s) { return ""; }

    public String zoo(Object o) { return ""; }

    public String nya(String s) { return ""; }
}

File: nj2k/testData/newJ2k/nullability/Overrides.kt
internal open class Base {
    open fun foo(s: String?): String? {
        return ""
    }

    open fun bar(s: String?): String? {
        return if (s != null) s + 1 else null
    }

    open fun zoo(o: Any?): String? {
        return ""
    }

    open fun nya(s: String): String {
        return s
    }
}

internal interface I {
    fun zoo(o: Any?): String?
    fun nya(s: String): String {
        return ""
    }
}

internal class C : Base(), I {
    override fun foo(s: String?): String? {
        return ""
    }

    override fun bar(s: String?): String? {
        return ""
    }

    override fun zoo(o: Any?): String? {
        return ""
    }

    override fun nya(s: String): String {
        return ""
    }
}
